pallets/account-manager/src/manager.rs

```diff
@@ -205,12 +226,14 @@ impl<T: Config> AccountManagerExt<T::AccountId, BalanceOf<T>, T::Hash, T::BlockN
         maybe_actual_fees: Option<BalanceOf<T>>,
     ) {
         if PendingChargesPerRound::<T>::get(T::Clock::current_round(), charge_id).is_some() {
-            <Self as AccountManagerExt<T::AccountId, BalanceOf<T>, T::Hash, T::BlockNumber>>::finalize(
-                charge_id,
-                outcome,
-                maybe_recipient,
-                maybe_actual_fees
-            ).expect("Expect try finalize to be infallible");
+            <Self as AccountManagerExt<
+                T::AccountId,
+                BalanceOf<T>,
+                T::Hash,
+                T::BlockNumber,
+                <T::Assets as Inspect<T::AccountId>>::AssetId,
+            >>::finalize(charge_id, outcome, maybe_recipient, maybe_actual_fees)
+            .expect("Expect try finalize to be infallible");
```

Comment:
`try` being infallible is a bit of misnomer 🙊

Reply:
yea I suggest we fix this part as Circuit rewrite to state machine